### ISSUE TITLE
Any slug that starts with `en` fails with Netlify adapter

### ISSUE THREAD

**Issue by epompeii** [withastro/astro#10806]
### Astro Info

```block
Astro                    v4.3.2
Node                     v20.12.2
System                   macOS (arm64)
Package Manager          npm
Output                   hybrid
Adapter                  @astrojs/netlify
Integrations             @astrojs/sitemap
                         astro-expressive-code
                         @astrojs/mdx
                         @astrojs/partytown
                         @astrojs/solid-js
```

Netlify plugin installed with this command:
```
npm install @astrojs/netlify@5.0.0 && npx astro add netlify -y 
```

### Describe the Bug

Any server side rendered route given a slug that starts with `en` fails with the Netlify adapter with a `502` error.
Here are some examples:
- https://bencher.dev/perf/en
- https://bencher.dev/perf/encrypted-message

This does not happen when testing locally or using the Node.js adapter.
It seems to only occur with the Netlify adapter.

I am using i18n support with Astro, and `en` is the default locale, so I think this has something to do with that.
Trying any other supported (non-default) locale, such as `fr` works just fine: https://bencher.dev/perf/fr

My i18n config looks like this:

```
i18n: {
    defaultLocale: 'en',
    locales: ['de', 'en', 'es', 'fr', 'ja', 'ko', 'pt', 'ru', 'zh'],
    routing: {
      prefixDefaultLocale: false,
    },
  },
 ```

Sometimes when hitting these endpoints (but not always) I get an error like this in the Netlify logs:

```
Apr 16, 08:24:16 PM: beb43464 ERROR  Invoke Error 	{"errorType":"TypeError","errorMessage":"fetch failed","cause":{"errorType":"Error","errorMessage":"C09F453CC37F0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:ssl/record/rec_layer_s3.c:1590:SSL alert number 80\n","code":"ERR_SSL_TLSV1_ALERT_INTERNAL_ERROR","library":"SSL routines","reason":"tlsv1 alert internal error","stack":["Error: C09F453CC37F0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:ssl/record/rec_layer_s3.c:1590:SSL alert number 80",""]},"stack":["TypeError: fetch failed","    at Object.fetch (node:internal/deps/undici/undici:11731:11)","    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","    at async #renderError (file:///var/task/node_modules/astro/dist/core/app/index.js:352:27)","    at async Module.handler (file:///var/task/node_modules/@astrojs/netlify/dist/ssr-function.js:29:30)","    at async Runtime.handler (file:///var/task/___netlify-bootstrap.mjs:1920:15)","    at async Runtime.handleOnceStreaming (file:///var/runtime/index.mjs:1206:26)"]}
```

There is a tracking issue for this here, if that is of interest: https://github.com/bencherdev/bencher/issues/369

### What's the expected result?

Any server side rendered route given a slug that starts with `en` should work just like any other slug.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/astro-i18n-en-netlify?file=astro.config.mjs

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by rishi-raj-jain** [comment]
To which URL do you expect the blog map to?

**Comment by epompeii** [comment]
This is fixed in Astro version `4.6.0` with `@astrojs/netlify@5.2.0`.
